mmcv/runner/hooks/lr_updater.py

```diff
@@ -62,6 +64,9 @@ def before_run(self, runner):
         self.base_lr = [
             group['initial_lr'] for group in runner.optimizer.param_groups
         ]
+        if self.warmup_byepoch:
+            assert runner.epoch_len
+            self.warmup_iters *= runner.epoch_len
```

Comment:
This can be implemented in `before_train_epoch`, where we can obtain the epoch_len by `len(runner.data_loader)`.